pages/account/index.vue

```diff
@@ -0,0 +1,163 @@
+                  <span class="px-3 py-2 text-zinc-400 text-sm">{{ user?.username }}</span>
+                </div>
+              </div>
+
+              <div>
+                <label class="block text-sm font-medium text-zinc-300">Display Name</label>
+                <input
+                  type="text"
+                  v-model="displayName"
+                  class="mt-1 block w-full rounded-md border-zinc-700 bg-zinc-800 text-zinc-100 shadow-sm focus:border-blue-500 focus:ring-blue-500 sm:text-sm"
+                />
+              </div>
+
+              <div>
+                <label class="block text-sm font-medium text-zinc-300">Email Address</label>
+                <input
+                  type="email"
+                  v-model="email"
+                  class="mt-1 block w-full rounded-md border-zinc-700 bg-zinc-800 text-zinc-100 shadow-sm focus:border-blue-500 focus:ring-blue-500 sm:text-sm"
+                />
+              </div>
+
+              <div>
+                <label class="block text-sm font-medium text-zinc-300">Account Type</label>
+                <div class="mt-1 flex rounded-md bg-zinc-800/50 shadow-sm ring-1 ring-inset ring-zinc-700">
+                  <span class="px-3 py-2 text-zinc-400 text-sm">{{ user?.admin ? 'Administrator' : 'User' }}</span>
+                </div>
+              </div>
+            </div>
+
+            <!-- Save Button -->
+            <div class="flex justify-end pt-4">
+              <button
+                type="button"
+                :disabled="!hasChanges"
+                @click="saveChanges"
+                class="rounded-md bg-blue-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-blue-500 disabled:opacity-50 disabled:cursor-not-allowed"
+              >
+                Save Changes
+              </button>
+            </div>
+          </div>
+        </div>
+
+        <!-- Security Section -->
+        <div class="bg-zinc-800 shadow-lg shadow-zinc-900/20 rounded-lg p-6">
+          <div class="pb-6 mb-6 border-b border-zinc-700">
+            <h3 class="text-lg font-medium leading-6 text-zinc-100">Security</h3>
+          </div>
+          <br>
+          <div class="space-y-4">
+            <button
+              type="button"
+              @click="showChangePasswordModal = true"
+              class="w-full rounded-md bg-zinc-700 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-zinc-600"
+            >
+              Change Password
+            </button>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+const user = useUser()
+
+// Form state
+const displayName = ref(user.value?.displayName || '')
+const email = ref(user.value?.email || '')
+
+// Track if form has changes
+const hasChanges = computed(() => {
+  return displayName.value !== user.value?.displayName || 
+         email.value !== user.value?.email
+})
+
+// Save changes
+async function saveChanges() {
```

Comment:
This function shouldn't have a try-catch, instead using a `saveChanges_wrapper` that uses then, catch and finally. 

Also should use an error component that renders in the form, rather than a modal. Check other forms (like signin or admin metadata import) for an example.